src/ATen/native/xpu/NestedTensorTransformerFunctions.cpp

```diff
@@ -0,0 +1,109 @@
+    // Instead of erroring, call the generic version
+    if (!(padded.dim() == 4 && do_transform_0213) &&
+        !(padded.dim() == 3 && !do_transform_0213)) {
+      return at::native::nested_from_padded_generic(
+          padded, sizes, do_transform_0213);
+    }
+    if (padded.dtype() != at::kFloat && padded.dtype() != kHalf) {
+      TORCH_WARN_ONCE(
+          "nested_from_padded CUDA kernels only support fp32/fp16; falling "
+          "back to slower generic kernel");
+      return at::native::nested_from_padded_generic(
+          padded, sizes, do_transform_0213);
+    }
+    Tensor target_offsets =
+        at::native::NestedTensor_batch_offsets_from_size_tensor(sizes, 0);
+    Tensor padded_sizes_tensor = at::tensor(padded.sizes());
+    Tensor output = at::empty({padded_tensor_numel(sizes)}, padded.options());
+    Tensor target_size_sizes = sizes.reshape(-1);
+
+    target_offsets = target_offsets.to(at::Device(kXPU), at::kInt);
+    padded_sizes_tensor = padded_sizes_tensor.to(at::Device(kXPU), at::kInt);
+    target_size_sizes = target_size_sizes.to(at::Device(kXPU), at::kInt);
+
+    auto output_size_ptr = target_size_sizes.data_ptr<int>();
+    auto input_size_ptr = padded_sizes_tensor.data_ptr<int>();
+    auto offsets_ptr = target_offsets.data_ptr<int>();
+
+    Tensor padded_contiguous = padded.contiguous();
+
+    if (padded.dtype() == at::kFloat) {
+      if (do_transform_0213) {
+        xpu::remove_padding_transform0213_kernelLauncher(
+            padded_contiguous.data_ptr<float>(),
+            output.data_ptr<float>(),
+            offsets_ptr,
+            input_size_ptr,
+            output_size_ptr,
+            padded_contiguous.dim() - 2,
+            padded_contiguous.sizes()[0]);
+      } else {
+        xpu::remove_padding_kernelLauncher(
+            padded_contiguous.data_ptr<float>(),
+            output.data_ptr<float>(),
+            offsets_ptr,
+            input_size_ptr,
+            output_size_ptr,
+            padded_contiguous.dim() - 1,
+            padded_contiguous.sizes()[0]);
+      }
+    } else if (padded.dtype() == at::kHalf) {
+      if (do_transform_0213) {
+        xpu::remove_padding_transform0213_kernelLauncher(
+            padded_contiguous.data_ptr<c10::Half>(),
+            output.data_ptr<c10::Half>(),
+            offsets_ptr,
+            input_size_ptr,
+            output_size_ptr,
+            padded_contiguous.dim() - 2,
+            padded_contiguous.sizes()[0]);
+      } else {
+        xpu::remove_padding_kernelLauncher(
+            padded_contiguous.data_ptr<c10::Half>(),
+            output.data_ptr<c10::Half>(),
+            offsets_ptr,
+            input_size_ptr,
+            output_size_ptr,
+            padded_contiguous.dim() - 1,
+            padded_contiguous.sizes()[0]);
+      }
+    } else {
+      AT_ERROR("Only support fp32/fp16 for padded input");
+    }
+    return at::detail::make_tensor<at::native::NestedTensorImpl>(
```

Comment:
I remember we have not upstreamed `DispatchKey::NestedTensorXPU`. It should not work.

Reply:
@fengyuan14 , I created https://github.com/intel/torch-xpu-ops/issues/1141 to keep track of PRs related to `NestedTensor` support for xpu.